gateway/ln-gateway/src/db.rs

```diff
@@ -431,8 +431,7 @@ struct RegisteredIncomingContractKey(pub PaymentImage);
 pub struct RegisteredIncomingContract {
     pub federation_id: FederationId,
     /// The amount of the incoming contract, in msats.
-    /// TODO: Rename to `incoming_amount_msats` to be more explicit.
-    pub incoming_amount: u64,
```

Comment:
@joschisan why is this a u64 in the first place and not an `Amount`?

Reply:
Not sure why exactly here - generally we have some u64 msat amounts in the lnv2 module instead for easy interop with ln invoice amounts. Looks like in this case it should actually be an Amount though.